backend/experiment/management/commands/createexperiment.py

```diff
@@ -0,0 +1,122 @@
+
+        # Create a basic test file for the experiment
+        self.create_test_file(experiment_name)
+
+    def create_experiment_rule_class(self, experiment_name):
+        # Get the experiment name in different cases
+        experiment_name_snake_case, experiment_name_snake_case_upper, experiment_name_pascal_case = self.get_experiment_name_cases(experiment_name)
+
+        # Create a new file for the experiment class
+        filename = f"./experiment/rules/{experiment_name_snake_case}.py"
+
+        # Check if the file already exists
+        if os.path.isfile(filename):
+            # Warn the user that the file already exists and ask if they want to overwrite it
+            self.stdout.write(self.style.WARNING(f"File {filename} already exists"))
+            overwrite = input("Do you want to overwrite it? (y/n): ")
+
+            # If the user does not want to overwrite the file, exit the command
+            if overwrite.lower() != 'y':
+                self.stdout.write(self.style.WARNING(f"File {filename} was not created"))
+                return
+            else:
+                self.stdout.write(self.style.WARNING(f"File {filename} will be overwritten"))
+
+        # Create the file by copying ./experiment/management/commands/templates/experiment.py
+        with open(filename, 'w') as f:
+            with open('./experiment/management/commands/templates/experiment.py', 'r') as template:
+                f.write(template.read()
+                    .replace('NewExperiment', experiment_name_pascal_case)
+                    .replace('new_experiment', experiment_name_snake_case)
+                    .replace('NEW_EXPERIMENT', experiment_name_snake_case_upper)
+                    .replace('New Experiment', experiment_name.title())
+                )
+
+        self.stdout.write(self.style.SUCCESS(f"Created {filename} for experiment {experiment_name}"))
+
+    def register_experiment_rule(self, experiment_name, file_path):
+
+        # Get the experiment name in different cases
+        experiment_name_snake_case, experiment_name_snake_case_upper, experiment_name_pascal_case = self.get_experiment_name_cases(experiment_name)
+
+        # New lines to add
+        new_import = f"from .{experiment_name_snake_case} import {experiment_name_pascal_case}\n"
+        new_dict_entry = f"    {experiment_name_pascal_case}.ID: {experiment_name_pascal_case},\n"
+
+        with open(file_path, 'r') as file:
+            lines = file.readlines()
+
+        # Find the line to insert the new import, maintaining alphabetical order
+        import_index = next(i for i, line in enumerate(lines) if line.startswith('from .') and line > new_import)
+        lines.insert(import_index, new_import)
+
+        # Find the line to insert the new dictionary entry, maintaining alphabetical order within the EXPERIMENT_RULES
+        dict_start_index = lines.index("EXPERIMENT_RULES = {\n") + 1  # Start after the opening brace
```

Comment:
NB: I was hoping to supersede the `EXPERIMENT_RULES` dict with the solution proposed in #732

Reply:
I can make an issue to remove this method after #732 is merged? Or should I remove it here first?